File: test/cpp/case16_instance_member.cpp
// このファイルは `test/py/case16_instance_member.py` のテスト/実装コードです。
// 役割が分かりやすいように、読み手向けの説明コメントを付与しています。
// 変更時は、既存仕様との整合性とテスト結果を必ず確認してください。


#include "cpp_module/py_runtime.h"

// このファイルは `test/py/case16_instance_member.py` のテスト/実装コードです。
// 役割が分かりやすいように、読み手向けの説明コメントを付与しています。
// 変更時は、既存仕様との整合性とテスト結果を必ず確認してください。

struct Point {
    int64 x;
    int64 y;
    
    Point(int64 x, int64 y) {
        this->x = x;
        this->y = y;
    }
    int64 total() {
        return this->x + this->y;
    }
};

int main() {
    Point p = Point(2, 5);
    py_print(p.total());
    return 0;
}

File: test/rs/case16_instance_member.rs
#[path = "../../src/rs_module/py_runtime.rs"]
mod py_runtime;
use py_runtime::{math_cos, math_exp, math_sin, math_sqrt, perf_counter, py_bool, py_grayscale_palette, py_in, py_isalpha, py_isdigit, py_len, py_print, py_save_gif, py_slice, py_write_rgb_png};

// このファイルは自動生成です（native Rust mode）。

#[derive(Clone)]
struct Point {
    x: i64,
    y: i64,
}

impl Point {
    fn new(x: i64, y: i64) -> Self {
        let mut self_obj = Self {
            x: 0,
            y: 0,
        };
        self_obj.x = x;
        self_obj.y = y;
        self_obj
    }

    fn total(&mut self) -> i64 {
        return ((self.x) + (self.y));
    }
}

fn main() {
    let mut p: Point = Point::new(2, 5);
    py_print(p.total());
}
